### DISCUSSION TITLE
How to limit post or get speed

### DISCUSSION THREAD

**Discussion by diandengshi** [aio-libs/aiohttp#8439] [General]
###    I imagine limiting the speed of uploading or downloading files using get and post when my internet environment is not good. 
###  Just like how aioftp provides read_speed_limit and writes_speed_limit parameters to limit bandwidth.

```
async with aiohttp.ClientSession(connector=connector, timeout=timeout) as session:
   async with session.get(url) as response:
 	  if response.status == 200:
 		  async with AsyncLimiter(max_rate=1024):
			  async with aiofiles.open(file_path, 'wb') as local_file:
 				  chunk = await response.content.read(buffer_size)
 				  if not chunk:
 					  break
```
### AsyncLimiter not limit speed

**Comment by ratijas** [comment]
Looking at [aiolimiter](https://aiolimiter.readthedocs.io/en/stable/) docs, I think it was ever supposed to work like that. The limiter controls how often you can _enter_ a limited region; not any particular IO bandwidth or anything that happens _within_ the region itself.

In the code you posted there is no loop, meaning you'd only read a single chunk of `buffer_size` length, and that's it.

Instead, you could limit how often chunks are read/written. You'd use a loop, and inside that loop put the AsyncLimiter region, and only then inside the limited region read a chunk.
